### ISSUE TITLE
Uploading a .crx file seems broken

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#10936]
Looks like the code to convert a `.crx` to a `.xpi` is broken in prod:

https://sentry.prod.mozaws.net/operations/olympia-prod/issues/5363163/

```
ValueError: must have exactly one of create/read/write/append mode
(13 additional frame(s) were not displayed)
...
  File "olympia/devhub/views.py", line 561, in upload
    is_standalone=is_standalone, channel=channel)
  File "olympia/devhub/views.py", line 520, in handle_upload
    automated_signing=automated_signing, addon=addon, user=user)
  File "olympia/files/models.py", line 592, in from_post
    upload.add_file(chunks, filename, size)
  File "olympia/files/models.py", line 577, in add_file
    hash_func = write_crx_as_xpi(chunks, loc)
  File "olympia/files/utils.py", line 1147, in write_crx_as_xpi
    with tempfile.NamedTemporaryFile('rwb+', dir=settings.TMP_PATH) as tmp:

ValueError: must have exactly one of create/read/write/append mode
```

**Comment by diox** [comment]
Fun: there is a test, it never worked because it's placed in a class that tests `File`, not `FileUpload`. So it always pass doing almost nothing.

**Comment by AlexandraMoga** [comment]
@diox I've downloaded several .crx packages from chrome webstore and then submitted them on -dev. There is no 500 error generated anymore, and the package is being validated:

![image](https://user-images.githubusercontent.com/31961530/54613680-cba02a80-4a63-11e9-87a0-4550459b2280.png)

I've run into validation issues with addons that I've packed myself using `chrome://extensions/`. 
Here is a validation failure + the addon that generated it:
https://addons-dev.allizom.org/en-US/developers/upload/7662fa9cb6994bf38c8e4982452ab8a1
Extension: https://drive.google.com/open?id=1FdjboGrl79-j8my3pm9Z1lb9xPWbwGk5

I've also run the .crx through https://www.extensiontest.com/ which brings forth the following error:
`Failed to unpack file.`

Giving the fact that only the extensions packed through `chrome://extensions/` are failing, could suggest that this issue doesn't come from our end. 
However, the error message that we are generating is not very helpful.
